### ISSUE TITLE
Typing errors

### ISSUE THREAD

**Issue by simonpeters** [orizens/ngx-youtube-player#4]
Hi,

I used the angular cli to create a new project and included ng2-youtube-player. But on compiling I get this error.

ERROR in [default] /.../node_modules/ng2-youtube-player/src/youtube-player.component.d.ts:1:0 
Cannot find type definition file for 'youtube'.

ERROR in [default] /.../node_modules/ng2-youtube-player/src/youtube-player.component.d.ts:10:24 
Cannot find namespace 'YT'.

I'm not sure how to proceed. Does anyone know a solution for this?

**Comment by cbjerg** [comment]
run npm i @types/youtube and everything should work

**Comment by dineshk16feb** [comment]
I have one more issues similar like that after installing @type/youtube,

"Namespace 'YT' has no exported member 'EventArgs' "

**Comment by orizens** [comment]
please use version 0.0.28 for now.
```npm i @types/youtube@0.0.28```

**Comment by orizens** [comment]
hi - i upgraded the package to "@types/youtube": "0.0.29"

**Comment by ghost** [comment]
i got the same issue 
node_modules/ng2-youtube-player/src/models.d.ts(6,30): error TS2694: Namespace 'YT' has no exported member 'PlayerEvent'.
please somebody help

**Comment by ghost** [comment]
fire this below commend 
npm install @types/youtube --save

problem resolved.its work for me

**Comment by ghost** [comment]
npm install @types/youtube --save

**Comment by orizens** [comment]
Please install latest version

**Comment by kukac7** [comment]
the latest version still gives the issue:

```
[at-loader] Checking finished with 16 errors
Error in bail mode: [at-loader] ./node_modules/ng2-youtube-player/src/models.d.ts:1:1 
    TS2688: Cannot find type definition file for 'youtube'.```

**Comment by alex-samuilov** [comment]
I use version 0.0.30 and have the same errors:
Cannot find namespace 'YT'.
Cannot find name 'YT'.

**Comment by lucienimmink** [comment]
Using 0.0.30 or 0.0.41


ERROR in node_modules/ngx-youtube-player/src/models.d.ts(5,26): error TS2503: Cannot find namespace 'YT'.
node_modules/ngx-youtube-player/src/models.d.ts(6,27): error TS2503: Cannot find namespace 'YT'.
node_modules/ngx-youtube-player/src/models.d.ts(1,23): error TS2688: Cannot find type definition file for 'youtu
be'.
node_modules/ngx-youtube-player/src/services/youtube-player.service.d.ts(10,24): error TS2503: Cannot find names
pace 'YT'.
node_modules/ngx-youtube-player/src/services/youtube-player.service.d.ts(15,110): error TS2503: Cannot find name
space 'YT'.
node_modules/ngx-youtube-player/src/services/youtube-player.service.d.ts(16,18): error TS2503: Cannot find names
pace 'YT'.
node_modules/ngx-youtube-player/src/services/youtube-player.service.d.ts(17,19): error TS2503: Cannot find names
pace 'YT'.
node_modules/ngx-youtube-player/src/services/youtube-player.service.d.ts(18,35): error TS2503: Cannot find names
pace 'YT'.
node_modules/ngx-youtube-player/src/services/youtube-player.service.d.ts(19,23): error TS2503: Cannot find names
pace 'YT'.
node_modules/ngx-youtube-player/src/services/youtube-player.service.d.ts(20,113): error TS2503: Cannot find name
space 'YT'.
node_modules/ngx-youtube-player/src/services/youtube-player.service.d.ts(21,30): error TS2503: Cannot find names
pace 'YT'.
node_modules/ngx-youtube-player/src/services/youtube-player.service.d.ts(1,23): error TS2688: Cannot find type d
efinition file for 'youtube'.
node_modules/ngx-youtube-player/src/modules/youtube-player.component.d.ts(17,17): error TS2503: Cannot find name
space 'YT'.
node_modules/ngx-youtube-player/src/modules/youtube-player.component.d.ts(18,25): error TS2503: Cannot find name
space 'YT'.
node_modules/ngx-youtube-player/src/modules/youtube-player.component.d.ts(19,26): error TS2503: Cannot find name
space 'YT'.
node_modules/ngx-youtube-player/src/modules/youtube-player.component.d.ts(1,23): error TS2688: Cannot find type
definition file for 'youtube'.

**Comment by orizens** [comment]
@lucienimmink install ```npm i @types/youtube```

**Comment by lucienimmink** [comment]
awesome, might as well put that in the docs somewhere!

**Comment by AllefDouglasDev** [comment]
WHAT I DID?
First: run `npm i @types/youtube`
Secund: run `npm i`
and finaly:
In the package .\node_modules\ng2-youtube-player\src\modules has youtube-player.component.d.ts
or  .\node_modules\ngx-youtube-player\src\modules has youtube-player.component.d.ts
Change line 18 to :  `change: EventEmitter<YT.PlayerState>;`
If has more files with "`YT.PlayerEvent`", change to` YT.PlayerState`

**Comment by simeyla** [comment]
This is still confusing the heck out of me. In the end adding `youtube` to types solved it for me and either `ng build` or `ng serve` worked fine:

```
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/app",
    "module": "es2015",
    "types": ["youtube"]
  },
  "exclude": [
    "src/test.ts",
    "**/*.spec.ts"
  ]
}
```

Without this I was able to get the types installed for Visual Studio Code (nothing was underlined), but on compilation I'd get errors about YT being undefined.

@orizens I see that you did some crazy things in the 6.0 update such as calling `win()['YT']` manually. Were you just as confused as me or was there a reason you didn't add `"youtube"` to types.

**Comment by orizens** [comment]
@simeyla actually, in v6, "@types/youtube" is now set as a dependency. 
I was wondering to move it into peerDepdencies. any suggestion?

**Comment by simeyla** [comment]
@orizens I'm not really an expert on how everything fits together. I just got the impression that by doing `win()['YT']` you were trying to solve a problem you couldn't find another solution to. I made my own control and looked at your code to see how you were fixing the problem of `YT` not being defined at compile time even though my editor could recognize it. Adding `"types": ["youtube"]` to the compilerOptions fixed that for me. I guess I don't understand how installing a types package alone isn't enough to make the compiler recognize it, but I was going round in circles trying to find an explanation.
